Source/missiles.cpp

```diff
@@ -412,17 +420,18 @@ void CheckMissileCol(Missile &missile, int minDamage, int maxDamage, bool isDama
 	bool blocked = false;
 	const int8_t pid = dPlayer[mx][my];
 	if (pid > 0) {
+		Player &hitPlayer = Players[pid - 1];
 		if (missile._micaster != TARGET_BOTH && !missile.IsTrap()) {
-			if (missile._micaster == TARGET_MONSTERS) {
-				if ((pid - 1) != missile._misource)
-					isPlayerHit = Plr2PlrMHit(Players[missile._misource], pid - 1, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted, &blocked);
+			if (missileSource == MissileSource::Player) {

```

Comment:
This looks like another switch-case for Player, Monster, Trap

Reply:
I tried to convert it, but the first `else` happens when the _micaster is TARGET_BOTH or a trap

Reply:
Here is what I came up with:

```cpp
		pid -= 1;
		int earflag = 0;
		switch (missileSource) {
		case MissileSource::Player: {
			Player *player = missile.sourcePlayer();
			if (&Players[pid] != player) {
				isPlayerHit = Plr2PlrMHit(*player, pid, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted, &blocked);
				break;
			}
			earflag = IsAnyOf(missile._miAnimType, MFILE_FIREWAL, MFILE_LGHNING) ? 1 : 0;
		} // No Break
		default: {
			Monster *monster = missileSource == MissileSource::Monster ? missile.sourceMonster() : nullptr;
			isPlayerHit = PlayerMHit(pid, monster, missile._midist, minDamage, maxDamage, missile._mitype, isDamageShifted, earflag, &blocked);
			break;
		}
		}
```

It's not much of a switch, but it is using it to do less comparisons and combine some calls.